graphix/instruction.py

```diff
@@ -29,181 +32,264 @@ class InstructionKind(enum.Enum):
     ZC = "ZC"
 
 
-class Instruction(BaseModel, abc.ABC):
+# Decorator required
+@dataclasses.dataclass
+class Instruction(ABC):
     """
-    Circuit instruction base class model.
+    Circuit instruction base class.
     """
 
-    kind: InstructionKind = None
-    meas_index: int = None
+    @property
+    @abc.abstractmethod
+    def kind(self) -> Any: ...
```

Comment:
The same comments as for `Command` apply for `Instruction`:
* `kind` should be of type `InstructionKind`.
* `kind` should be a field.
* The class `Instruction` should be renamed (for instance, as `BaseInstruction`) and the type `Instruction` should be a union type.

Reply:
Fixed in d4a15c0e4de2c2ffe5cef8d6afff2c03f75eb0c9 .
As we need to very careful when defining type aliases (leads to `mypy` warning if not exhaustive/incorrect), I temporally dropped some of them.